### PR TITLE
Tests to validate autotune layer config object id

### PR DESCRIPTION
**Pull request by shruacha1234**
Signed-off-by: Shruthi <Shruthi.Acharya@ibm.com>

### CODE REVIEW

**Comment by chandrams** [comment]
@shruacha1234 - can you update the readme too.

File: tests/README.md [hunk 1]
```diff
@@ -40,9 +40,39 @@
...
+  
+- **Autotune object id tests**
+
+  Here we validate the autotune object id for different scenarios.
+  The test does the following:
+  - Deploys autotune and its dependencies using the deploy script from the autotune repo
+  - Deploys benchmark applications and requried application autotune yamls
+  - Validate autotune id for following scenarios:
+  	1. Check the uniqueness of the autotune object ids
+  	2. Check if re-applying the autotune object without modifying yaml changes the autotune object id
```
Comment (chandrams) [inline tests/README.md:61]: Change this to - Check if re-applying the autotune object without modifying the yaml does not change the autotune object id

File: tests/autotune_test_yamls/manifests/da/autotune_config_id_test_yaml/new_layer_config_test/openj9-config.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+layer_name: openj9
+layer_level: 0
+details: java openj9 tunables
+layerPresence:
+  # This layer is assumed to be available by default for all applications.
+  presence: always
+tunables:
+  - name: javaHeap
+    value_type: double
+    upper_bound: '250M'
```
Comment (chandrams) [inline tests/autotune_test_yamls/manifests/da/autotune_config_id_test_yaml/new_layer_config_test/openj9-config.yaml:14]: Upper bound and lower bound are now defined as double

File: tests/README.md [hunk 3]
```diff
@@ -40,9 +40,39 @@
...
   - Deploys autotune and its dependencies using the deploy script from the autotune repo
   - Modify the layer config and apply
-  - Validate if the modified config is reflected in the JSON output from listAutotuneTunables Autotune API  
+  - Validate if the modified config is reflected in the JSON output from listAutotuneTunables Autotune API
 
+- **ConfigMap yaml tests**
 
+  Here we modify the configmap yaml and validate if the behaviour is reflected in autotune. We also check if autotune identifies invalid configurations well by giving out appropriate error messages to the users.
+  The test does the following:
+  - Modify the configmap yaml
```
Comment (chandrams) [inline tests/README.md:49]: You can add the scenario details here

**Review by chandrams — CHANGES_REQUESTED** [review]

File: tests/README.md [hunk 4]
```diff
@@ -36,13 +36,49 @@
...
+  - Deploys autotune and its dependencies using the deploy script from the autotune repo
+  - Checks if autotune validates the configmap yaml
+  - Test validates the following scenarios:
+  	1. invalid-cluster-type
```
Comment (chandrams) [inline tests/README.md:53]: Start with caps

File: tests/README.md [hunk 5]
```diff
@@ -36,13 +36,49 @@
...
   - Deploys autotune and its dependencies using the deploy script from the autotune repo
   - Modify the layer config and apply
-  - Validate if the modified config is reflected in the JSON output from listAutotuneTunables Autotune API  
+  - Validate if the modified config is reflected in the JSON output from listAutotuneTunables Autotune API
 
+- **ConfigMap yaml tests**
 
+  Here we modify the configmap yaml and validate if the behaviour is reflected in autotune. We also check if autotune identifies invalid configurations well by giving out appropriate error messages to the users.
+  The test does the following:
+  - Modify the configmap yaml
```
Comment (chandrams) [inline tests/README.md:49]: Change Modify to Modifies

**Review by chandrams — APPROVED** [review]
LGTM

**Comment by dinogun** [comment]
Please resolve the conflicts for this PR

**Comment by dinogun** [comment]
LGTM

**Pull request merged**